kuksa-client/Dockerfile

```diff
@@ -8,23 +8,41 @@
 
 # Note: This dockerfile needs to be executed one level above in the root folder
 
-FROM python:3.10-alpine as build
-RUN apk update && apk add git alpine-sdk linux-headers
-RUN pip install --upgrade pip build
+
+FROM python:3.10-slim-bookworm as build
```

Comment:
Should we possibly document our strategy for Docker environments somewhere. Like, should we stop using alpine everywhere? Today we have a quite big mix, and are there not some possible benefits of using the same environment as often as possible?

Reply:
"probably", although I think it might differ depending on technology

So "historically" there was Alpine used, as for the C++ Val-Server it really made a difference size-wise.

For Python based things the savings are not that large, and using glibc builds are just easiest. Current stable debian just seems a common choice in many docker workloads. The solution here basically already includes learnings from the past , e.g. the pyinstaller technique from DBC feeder, except it does not condense all pyinstaller collected things in one file (as it does not offer benefits inside a container, but a runtime penalty unpacking it)

For rust it is a bit different, as we are using "from scratch" images (e.g. only  one binary in the container, no distro), and glibc isn't really suited to be statically linked  - you CAN do it, but then some rust crates/dependecies break that, hence we have MUSL there.

tldr; We should, I think this on represents our current "best practice" in pyhton, and maybe we should adapt our python stuff